### ISSUE TITLE
Bug: block-scoped-var rule incorrectly flags break/continue with label

### ISSUE THREAD

**Issue by aparajita** [eslint/eslint#3082]
Currently, the `block-scoped-var` rule generates false warnings for `break` and `continue` statements that use labels.

**Version:** v0.24.1

**Source:**
```js
/* eslint no-labels:0, block-scoped-var:2 */

foo:
for (var i = 0; i < 10; ++i) {
    bar:
    for (var j = 0; j < 13; ++j) {
        if (j === 7) {
            continue foo;  // warning
        }

        if (j === 2) {
            break bar;  // warning
        }
    }
}
```

**Config:** Invoked with `eslint --no-eslintrc --reset`, so the complete config is in the source config.

**Output:**
```
   8:21  error  "foo" used outside of binding context  block-scoped-var
  12:18  error  "bar" used outside of binding context  block-scoped-var
```

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by aparajita** [comment]
I'll resubmit this when I have all of the required info.

**Comment by aparajita** [comment]
I have a fix, will be submitting it soon.
